### PR TITLE
Impbk2002 patch immortal ambition

### PR DESCRIPTION
**Pull request by impbk2002**
Vaal Immortal Ambition to work properly as if it is a life leech
fixed 1
fixed 2
fixed 3
fixed 4

### CODE REVIEW

File: Modules/CalcOffence.lua [hunk 1]
```diff
@@ -2134,6 +2134,18 @@ function calcs.offence(env, actor, activeSkill)
 	output.ManaLeechInstanceRate = output.Mana * data.misc.LeechRateBase * calcLib.mod(skillModList, skillCfg, "ManaLeechRate")
 	output.ManaLeechRate = output.ManaLeechInstantRate + m_min(output.ManaLeechInstances * output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaRecoveryRateMod
 	output.ManaLeechPerHit = output.ManaLeechInstant + m_min(output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaLeechDuration * output.ManaRecoveryRateMod
+	
+	if skillModList:Flag(nil, "ImmortalAmbition") then
+		local val1 = output.EnergyShieldLeechInstanceRate 
+		local val2 = output.EnergyShieldLeechRate 
+		local val3 = output.EnergyShieldLeechPerHit
```
Comment (Regisle) [inline Modules/CalcOffence.lua:2141]: small nit. val 1 unused and not very descriptive or needed names

Comment (impbk2002) [reply Modules/CalcOffence.lua]: Got it, thx

**Comment by LocalIdentity** [comment]
Looks like you accidentally merged the changes from the flask PR you made into this one too. Please remove those changes from this PR

**Review by ppoelzl — COMMENTED** [review]
I will review your PRs tomorrow my time, but for general feedback:
* Please make your PRs against the dev branch, not master.
* Please use meaningful and expressive variable names (manaInstantRecovery, not val1, val2, etc.)
* Please avoid excessive whitespace (probably all empty lines you've added fall into this category) and other formatting not consitent with the surrounding code (add whitespace after commas, avoid whitespace before and after round brackets, etc.

File: Modules/CalcOffence.lua [hunk 2]
```diff
@@ -2134,6 +2134,13 @@ function calcs.offence(env, actor, activeSkill)
 	output.ManaLeechInstanceRate = output.Mana * data.misc.LeechRateBase * calcLib.mod(skillModList, skillCfg, "ManaLeechRate")
 	output.ManaLeechRate = output.ManaLeechInstantRate + m_min(output.ManaLeechInstances * output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaRecoveryRateMod
 	output.ManaLeechPerHit = output.ManaLeechInstant + m_min(output.ManaLeechInstanceRate, output.MaxManaLeechRate) * output.ManaLeechDuration * output.ManaRecoveryRateMod
+	-- Immortal Ambition kicks in as if it is a energy shield leech
```
Comment (ppoelzl) [inline Modules/CalcOffence.lua:2137]:
```suggestion
	-- On full life, Immortal Ambition treats life leech as energy shield leech
```

File: Modules/CalcOffence.lua [hunk 3]
```diff
@@ -2134,6 +2134,13 @@ function calcs.offence(env, actor, activeSkill)
...
+	if skillModList:Flag(nil, "ImmortalAmbition") then
+		output.EnergyShieldLeechRate = output.EnergyShieldLeechRate + output.LifeLeechRate
+		output.EnergyShieldLeechPerHit = output.EnergyShieldLeechPerHit  + output.LifeLeechPerHit
+		-- to disable leechLife flag clear output.LifeLeechRate
```
Comment (ppoelzl) [inline Modules/CalcOffence.lua:2141]:
```suggestion
		-- Clears output.LifeLeechRate to disable leechLife flag
```

**Review by ppoelzl — COMMENTED** [review]
Ready to be merged after the mentioned changes to the documentation.

**Review by ppoelzl — APPROVED** [review]
LGTM now.

**Pull request merged**